providers/refresh.md

```diff
@@ -0,0 +1,140 @@
+
+The persister also defines what will be deleted as part of a refresh.  The saver will delete the compliment of what is in the inventory collection based on a certain scope.  During a full refresh this scope is the entire association (e.g. the equivalent of `ems.vms.all`) and during a targeted refresh the scope is basically the set of targets.
+
+The builtin collections come from the core [`ManageIQ::Providers::Inventory::Persister`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/manageiq/providers/inventory/persister.rb) classes.  Each manager type has their own set of collections defined by the `Persister::Builder` subclass, e.g. here is the default set for an [`InfraManager`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/manageiq/providers/inventory/persister/builder/infra_manager.rb).  Your provider plugin's persister class adds collections from this "menu" using the `add_collection` method.
+
+You can modify properties of these builtin collections by providing a block to the `add_collection` call, as well as add entirely new collection if you want.
+
+It is also possible to define collections with a "custom save block" where you can do anything that doesn't fit into the typical model saver.
+
+For a deep dive into the inventory collection options check out the [Inventory Collection](../persister/inventory_collections.md) documentation.
+
+###### Inventory::Parser
+
+The parser is where the majority of the refresh logic lives.  It is here that the data transformation between the native provider schema and the ManageIQ schema happens.
+
+This typically consists of changing property names to match MIQ columns, changing property units, and linking inventory together.
+
+The parser builds Inventory Objects in the different Inventory Collections which are then saved as records in the database by save_inventory.
+
+```ruby
+def vms
+  # Loop through the collected vms
+  collector.vms.each do |vm|
+
+    # Build an InventoryObject in the vms inventory collection
+    persister.vms.build(
+      :name    => vm.name,
+      :ems_ref => vm.id,  # Make sure that you populate all of the collection's
+                          # manager_ref attributes
+    )
+  end
+end
+```
+
+The Inventory Collections define a "manager_ref" which is a combination of columns which indicate uniqueness within a collection.  For most "top-level" models e.g. Vms, Hosts, etc... this is just the ems_ref column.
+
+The trickiest part of parsing is linking records together.  This is done by using the `#lazy_find` mechanism that the inventory_collections provide.  Essentially what this does is instruct the save_inventory code to resolve a query later when all dependencies are satisfied.  Dependencies are tracked by treating Inventory Objects as nodes and links between as edges in a directed-acyclic-graph.
```

Comment:
I'm not sure we need the nodes/edges/and graph info as it might be too much information at this high level.  I think it's sufficient to say that it sets up a lazy link that is resolved later, allowing for collection without regard for order.